src/test/store/tracks.test.js

```diff
@@ -0,0 +1,445 @@
+    );
+    const workerThreadIndex = profile.threads.findIndex(
+      thread => thread.name === 'DOM Worker'
+    );
+    const styleThreadIndex = profile.threads.findIndex(
+      thread => thread.name === 'Style'
+    );
+    const parentPid = profile.threads[parentThreadIndex].pid;
+    const tabPid = profile.threads[workerThreadIndex].pid;
+    const globalTracks = ProfileViewSelectors.getGlobalTracks(getState());
+    const parentTrackIndex = globalTracks.findIndex(
+      track =>
+        track.type === 'process' && track.mainThreadIndex === parentThreadIndex
+    );
+    const tabTrackIndex = globalTracks.findIndex(
+      track =>
+        track.type === 'process' && track.mainThreadIndex === tabThreadIndex
+    );
+    profile.threads.findIndex(
+      thread => thread.name === 'GeckoMain' && thread.processType === 'tab'
+    );
+    let styleTrackIndex, workerTrackIndex;
+    for (const [, tracks] of ProfileViewSelectors.getLocalTracksByPid(
+      getState()
+    )) {
+      for (let trackIndex = 0; trackIndex < tracks.length; trackIndex++) {
+        const track = tracks[trackIndex];
+        if (track.type === 'thread') {
+          switch (track.threadIndex) {
+            case styleThreadIndex:
+              styleTrackIndex = trackIndex;
+              break;
+            case workerThreadIndex:
+              workerTrackIndex = trackIndex;
+              break;
+            default:
+              break;
+          }
+        }
+      }
+    }
+    if (
+      typeof styleTrackIndex !== 'number' ||
+      typeof workerTrackIndex !== 'number'
+    ) {
+      throw new Error('A local track index could not be found');
+    }
+
+    return {
+      // Store:
+      dispatch,
+      getState,
+      // Indexes:
+      parentThreadIndex,
+      parentTrackIndex,
+      tabThreadIndex,
+      tabTrackIndex,
+      styleTrackIndex,
+      styleThreadIndex,
+      workerTrackIndex,
+      workerThreadIndex,
+      parentPid,
+      tabPid,
+    };
+  }
+
+  describe('global tracks', function() {
+    it('starts out with the initial sorting', function() {
+      const { getState } = init();
+      expect(getHumanReadableTracks(getState())).toEqual([
+        'show [thread GeckoMain process]',
+        'show [thread GeckoMain tab] SELECTED',
+        '  - show [thread DOM Worker]',
+        '  - show [thread Style]',
+      ]);
+    });
+
+    it('can hide a global track', function() {
+      const { getState, dispatch, tabTrackIndex } = init();
+      withAnalyticsMock(() => {
+        dispatch(hideGlobalTrack(tabTrackIndex));
+        expect(self.ga).toBeCalledWith('send', {
+          eventAction: 'hide global track',
+          eventCategory: 'timeline',
+          hitType: 'event',
+        });
+      });
+      expect(getHumanReadableTracks(getState())).toEqual([
+        'show [thread GeckoMain process] SELECTED',
+        'hide [thread GeckoMain tab]',
+        '  - show [thread DOM Worker]',
+        '  - show [thread Style]',
```

Comment:
In the state, it hides only the `GlobalTrack`, but in the UI the `LocalTracks` would not be shown, since they would be a child component.

This would be rendered something like:
```js
<GlobalTrackComponent />
<GlobalTrackComponent style={{display: none}}>
  <LocalTrackComponent />
  <LocalTrackComponent />
</GlobalTrackComponent>
```

Reply:
I see. Good for me for now :)